topdown/graphql.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2022 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package topdown
+
+import (
+	"fmt"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/topdown/builtins"
+	gqltop "github.com/vektah/gqlparser/v2"
+	gqlast "github.com/vektah/gqlparser/v2/ast"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+	gqlparser "github.com/vektah/gqlparser/v2/parser"
+	gqlvalidator "github.com/vektah/gqlparser/v2/validator"
+)
+
+// Parses a GraphQL schema, and returns only the Schema object.
+// Used in validation of queries.
+// Note that the error type here is a gqlerror.Error struct.
+func loadSchema(schema string) (*gqlast.Schema, error) {
+	loadedSchema, err := gqltop.LoadSchema(&gqlast.Source{Input: schema})
+	if err != nil {
+		return nil, err
+	}
+	return loadedSchema, nil
```

Comment:
Can you drop in a reference to the SA check? I know I'll keep wondering why this isn't
```go
return gqltop.LoadSchema(&gqlast.Source{Input: schema})
```
forever on... 🙃

Reply:
Absolutely! I'll add that shortly, along with the imports fixup.

Reply:
New comments include a link to the SA4023 docs page, along with extended note about the issue.